Tests/test_SeqIO_index.py

```diff
@@ -357,10 +353,28 @@ def tearDown(self):
         if os.path.isfile(self.index_tmp):
             os.remove(self.index_tmp)
 
+    @classmethod
+    def get_mode(cls, fmt):
+        mode = cls.modes.get(fmt)
+        if mode is not None:
+            return mode
+        streams = {"rt": StringIO(), "rb": BytesIO()}
+        for mode in streams:
```

Comment:
This seems overly complicated.

How about using ``for mode, stream in stream.items():`` here.

Or, replace the dict with a list or tuple of 2-tuples and do:

```python
for mode, stream in (("rt", StringIO()), ("rb", BytesIO())):
```